### PR TITLE
DOC: Improve documentation formatting

### PR DESCRIPTION
**Pull request by oscargus**
* Replaced `...` with `\dots`
* Fewer `*` used for multiplication in LaTeX
* Other minor fixes
* The standard editor-introduced trailing space removal

Draft as not all poly methods are yet handled (it would be really nice if the docstring could be inherited as mostly just the name of the polynomial differs...).

### CODE REVIEW

**Comment by mattip** [comment]
I am not sure about removing `*` from math equations. I think it reduces readability. What is the motivation for this?
```diff
- math:: p(x) = c_0 + c_1 * T_1(x) + ... +  c_n * T_n(x)
+ math:: p(x) = c_0 + c_1 T_1(x) + \\dots +  c_n T_n(x)
```

**Comment by oscargus** [comment]
It looks "bad" in LaTeX... #21310 (If anything it is convolution, not multiplication, see e.g. https://en.wikipedia.org/wiki/Convolution .) But I guess it is really down to when the doc-string is primarily used. Should it look good in the terminal (then one should maybe not even use math and subscripts etc) or in the rendered HTML/PDF-documentation?

It may be that I scrap this then, but I'd be a bit interested in getting feedback on using a common doc-string for similar methods, see the `_doc.py`-file and the example in `hermite_e.py`. This will make it easier to keep things consistent (and changes similar to those done here only need to be done once rather than six or seven times). But will make it harder to read the documentation while editing the files.

**Comment by oscargus** [comment]
Also, right now, the Hermite E-polynomials are referred to as Hermite, Hermite_e, and HermiteE in different functions. This can be avoided as well.

**Comment by oscargus** [comment]
Here is the outcome of the new doc-approach: https://output.circle-artifacts.com/output/job/9f6cbc0a-b81c-47d9-8f5c-6d55ebf945e7/artifacts/0/doc/build/html/reference/generated/numpy.polynomial.hermite_e.herme2poly.html#numpy.polynomial.hermite_e.herme2poly

**Comment by mattip** [comment]
[here](https://numpy.org/doc/stable/reference/generated/numpy.polynomial.chebyshev.chebfromroots.html) is the HTML rendered version of the current docs. The `*` symbol seems to render fine: it is not a different size or weight than the rest of the symbols there.

**Comment by mattip** [comment]
[here](https://output.circle-artifacts.com/output/job/9f6cbc0a-b81c-47d9-8f5c-6d55ebf945e7/artifacts/0/doc/build/html/reference/generated/numpy.polynomial.chebyshev.chebfromroots.html?highlight=chebfromroots#numpy.polynomial.chebyshev.chebfromroots) is the same page without the `*`. The symbols are rendered nicely without space between c_1 and T_1

**Comment by oscargus** [comment]
When I said "bad", I really meant that `*` is not the typical symbol to use for multiplication in LaTeX. Then, it can still be a decent choice if terminal rendering is prioritized.

File: doc/source/reference/arrays.ndarray.rst [hunk 1]
```diff
@@ -108,7 +108,7 @@ A segment of memory is inherently 1-dimensional, and there are many
 different schemes for arranging the items of an *N*-dimensional array
 in a 1-dimensional block. NumPy is flexible, and :class:`ndarray`
 objects can accommodate any *strided indexing scheme*. In a strided
-scheme, the N-dimensional index :math:`(n_0, n_1, ..., n_{N-1})`
+scheme, the N-dimensional index :math:`(n_0, n_1, \\dots, n_{N-1})`
```
Comment (eric-wieser) [inline doc/source/reference/arrays.ndarray.rst:111]: Why the double backslash?

File: numpy/lib/function_base.py [hunk 2]
```diff
@@ -2466,7 +2466,7 @@ def cov(m, y=None, rowvar=True, bias=False, ddof=None, fweights=None,
     Estimate a covariance matrix, given data and weights.
 
     Covariance indicates the level to which two variables vary together.
-    If we examine N-dimensional samples, :math:`X = [x_1, x_2, ... x_N]^T`,
+    If we examine N-dimensional samples, :math:`X = [x_1, x_2, \\dots, x_N]^T`,
```
Comment (eric-wieser) [inline numpy/lib/function_base.py:2469]: I'd err on the side of making this an `r""" ⋯ """` docstring rather than adding backslashes

**Comment by HaoZeke** [comment]
My two cents on the multiplication symbol would be to use `\times` which is rendered nicely by both mathjax and LaTeX.

**Comment by tgross35** [comment]
Came across this randomly and figured I'd chip in for completeness:
- `*` renders the same as `\ast` which is usually reserved for convolution. It's clear from context that it's not the case, but this bit from the demo page does kind of look like it could be explaining some sort of chained convolution property image
- Usually no symbol is used to display multiplication of variables. Example with both convolution and multiplication: 
image
- If a symbol is used for multiplication, it's usually `\cdot` for `a · b` or `ab`. This is also used for vector dot product and array multiplication, though it can often be omitted
image image
- `\times` is for vector cross product
image

General reference related to this is mentioned here https://latexref.xyz/Math-symbols.html

Of course, those are the conventions for when writing a document where you care about the display much more than the readability of the raw latex. IMO the rendered example without `*` looks much cleaner, and the `c_1 T_1(x)` vs. `c_1 * T_1(x)` is at least as easy to read as `\dots` (or any other latex builtins. I could see preference varying to end users who aren't latex-friendly though.

If user readability of the ascii version is considered valuable, there are markup alternatives like [asciimath](http://asciimath.org/) that allow writing in a way that is more readable raw than standard latex (`...` renders as `\dots`, `*` renders as `\cdot`, `(a+b)/c` as `\frac{a+b}{b}`, etc), and could be allowed with `:easymath:` or similar.

**Comment by seberg** [comment]
Can we make a call on this one?  My personal question would be if living in 2022 now, we cannot finally rely on unicode characters working correctly.  And thus, just use at least `⋅` where `*` would render badly, but adding `\cdot` everywhere seems not nice either.

**Comment by mattip** [comment]
@oscargus will you be continuing with this PR?

**Comment by oscargus** [comment]
Sorry, this completely slipped my mind... Will try to have a look over the weekend.

**Comment by mdhaber** [comment]
> we cannot finally rely on unicode characters working correctly

I think you can. We use many in SciPy, anyway.

**Comment by mdhaber** [comment]
Looks like this is no longer active and there are substantial merge conflicts, so I'll go ahead and close. If you're still interested, @oscargus, please reopen and ping me for review - I think these were worthwhile improvements, and I think it will be easier to push through if you use special characters instead of LaTeX.

**Pull request closed without merging**